src/tests/test_connection.py

```diff
@@ -92,3 +97,8 @@ def test_missing_driver(
             Connection.from_connect_str(connect_str)
 
         assert "try to install package: " + missing_pkg in str(error.value)
+
+
+def test_no_current_connection_and_get_info(mock_init_connection):
+    with pytest.raises(AttributeError):
+        Connection.current._get_curr_connection_info()
```

Comment:
shouldn't we be testing? that `Connection.current._get_curr_connection_info()` *doesnt fail* when current is None?

I think we have a leaky abstraction [here](https://github.com/ploomber/jupysql/blob/7b9f0f17c41342c31872708dd431633c7576b09b/src/sql/plot.py#L247), the if statement should be inside the `_get_curr_connection_info` and return None if there isn't an active connection. this way someone using the function doesn't have to know anything about the internals, as it's safe to call even if there are no connections

Reply:
I am thinking to declare the `_get_curr_connection_info` as classmethod might make much sense